core/vm/actor/builtin/v0/miner/miner_actor_state.cpp

```diff
@@ -16,8 +17,10 @@ namespace fc::vm::actor::builtin::v0::miner {
 
   outcome::result<types::miner::MinerInfo> MinerActorState::getInfo(
       IpldPtr ipld) const {
-    OUTCOME_TRY(info0, ipld->getCbor<MinerInfo>(miner_info));
-    return std::move(info0);
+    OUTCOME_TRY(info, ipld->getCbor<MinerInfo>(miner_info));
+    OUTCOME_TRYA(info.window_post_proof_type,
```

Comment:
Why do you not satisfied by value from ipld?

Reply:
RegisteredPoStProof type combines WinningPost and WindowPost and in v0 and v2 actors the WinningPost stored in IPLD miner state in 'window_post_proof_type'. But the function expects WindowPost, so Lotus determines WindowPost for SealProofType. There are much more code for actor in Lotus, see (https://github.com/filecoin-project/lotus/blob/master/chain/actors/builtin/miner/v0.go#L315). I don't know the reasoning for such things.

Reply:
Ok